vsc_pdf_template_transformer/lib/models/tpl_transform.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:pdf/widgets.dart';
+import 'package:vsc_pdf_template_transformer/models/tpl_alignment.dart';
+import 'package:vsc_pdf_template_transformer/models/tpl_pdf_point.dart';
+import 'package:vsc_pdf_template_transformer/utils/evaluator.dart';
+import 'package:vsc_pdf_template_transformer/utils/widget_builder.dart' as wb;
+import 'package:vector_math/vector_math_64.dart';
+import '../utils/widget_json_converter.dart';
+
+part 'tpl_transform.g.dart';
+
+@JsonSerializable(
+  checked: true,
+  disallowUnrecognizedKeys: false,
+  explicitToJson: true,
+)
+class TplTransform implements wb.WidgetBuilder {
+  TplTransform();
+
+  String className = 'TplTransform';
+  dynamic transform;
+  TplPdfPoint? origin;
+  TplAlignment? alignment;
+  dynamic adjustLayout;
+  dynamic unconstrained;
+  dynamic fit;
+  dynamic angle;
+  TplPdfPoint? offset;
+  dynamic scale;
+  @WidgetJsonConverter()
+  wb.WidgetBuilder? child;
+
+  factory TplTransform.fromJson(Map<String, dynamic> json) =>
+      _$TplTransformFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() => _$TplTransformToJson(this);
+
+  @override
+  Widget buildWidget(Map<String, dynamic> data) {
+    final transformStr = evaluateString(transform, data);
+    switch (transformStr) {
+      case 'rotate':
+        return Transform.rotate(
+            angle: angle,
+            origin: origin?.toPdf(data),
+            alignment: alignment?.buildAlignment(data),
+            child: child?.buildWidget(data));
+
+      case 'rotateBox':
+        return Transform.rotateBox(
+            angle: angle,
+            child: child?.buildWidget(data),
+            unconstrained: false);
+
+      case 'translate':
+        return Transform.translate(
+            offset: offset!.toPdf(data), child: child?.buildWidget(data));
+
+      case 'scale':
+        return Transform.scale(
+            scale: scale,
+            origin: origin?.toPdf(data),
+            alignment: alignment?.buildAlignment(data),
+            child: child?.buildWidget(data));
+    }
+    return Transform(
```

Comment:
This really doesn't do anything for us. What if I misspell `rotateBox` as `rotatBox`? It would just fall through to the default constructor and nothing would work. I would have no idea why. Let's throw an exception if the transformStr is not valid.